File: Android/melody-runtime/src/main/java/com/melody/runtime/components/MelodyText.kt
package com.melody.runtime.components

import androidx.compose.foundation.layout.wrapContentSize
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.sp
import com.melody.core.schema.ComponentDefinition
import com.melody.core.schema.ViewAlignment
import com.melody.core.schema.resolved
import com.melody.runtime.renderer.LocalThemeColors

@Composable
fun MelodyText(
    definition: ComponentDefinition,
    resolvedText: String
) {
    val themeColors = LocalThemeColors.current
    val style = definition.style

    val fontSize = (style?.fontSize.resolved ?: 16.0).sp
    val fontWeight = resolveFontWeight(style?.fontWeight)
    val fontFamily = resolveFontFamily(style?.fontDesign)
    val color = style?.color.resolved?.let { parseColor(it, themeColors) }
    val textAlign = when (style?.alignment.resolved) {
        ViewAlignment.Center -> TextAlign.Center
        ViewAlignment.Trailing, ViewAlignment.Right -> TextAlign.End
        else -> TextAlign.Start
    }
    val maxLines = style?.lineLimit.resolved ?: Int.MAX_VALUE

    Text(
        text = resolvedText,
        fontSize = fontSize,
        fontWeight = fontWeight,
        fontFamily = fontFamily,
        color = color ?: androidx.compose.ui.graphics.Color.Unspecified,
        textAlign = textAlign,
        maxLines = maxLines,
        overflow = TextOverflow.Ellipsis,
        modifier = Modifier.melodyStyle(style)
    )
}

File: Sources/Runtime/Components/MelodyText.swift
import SwiftUI
import Core

/// Renders a styled text label from a resolved string value.
struct MelodyText: View {
    let definition: ComponentDefinition
    let resolvedText: String
    @Environment(\.themeColors) private var themeColors

    var body: some View {
        Text(resolvedText)
            .font(.system(size: CGFloat(definition.style?.fontSize.resolved ?? 16),
                          weight: StyleResolver.fontWeight(definition.style?.fontWeight),
                          design: StyleResolver.fontDesign(definition.style?.fontDesign)))
            .foregroundStyle(textColor)
            .multilineTextAlignment(textAlignment)
            .lineLimit(definition.style?.lineLimit.resolved)
            .melodyStyle(definition.style)
    }

    private var textAlignment: TextAlignment {
        switch definition.style?.alignment?.literalValue {
        case .center: return .center
        case .trailing, .right: return .trailing
        default: return .leading
        }
    }

    private var textColor: Color {
        StyleResolver.color(from: definition.style, default: .primary, themeColors: themeColors)
    }
}
